src/_guides/environment-declarations.md

```diff
@@ -0,0 +1,188 @@
+```
+
+To learn more about specifying environment declarations,
+including for other tools and compilers, see
+[Specifying environment declarations][].
+
+[`dart run`]: /tools/dart-run
+[`dart compile`]: /tools/dart-compile
+[Specifying environment declarations]: #specifying-environment-declarations
+
+## Accessing environment declarations
+
+To access specified environment declaration values,
+use one of the `fromEnvironment` constructors
+with `const` or within a constant context.
+Use [`bool.fromEnvironment`][bool-from] for `true` or `false` values,
+[`int.fromEnvironment`][int-from] for integer values,
+and [`String.fromEnvironment`][string-from] for anything else.
+
+{{site.alert.note}}
+  The environment declaration constructors are only guaranteed
+  to work when invoked as `const`.
+  Most compilers must be able to evaluate their value at compile time.
+{{site.alert.end}}
+
+Each of the `fromEnvironment` constructors require the
+name or key of the environment declaration key-value pair,
+as well as an optional `defaultValue` override
+that is used when a declaration wasn't defined
+or the specified value cannot be parsed as the expected type.
+
+For example, if you only want to log messages
+when the environment declaration `DEBUG` is set to `true`:
+
+<?code-excerpt "misc/lib/development/environment_declarations.dart (debug-log)"?>
+```dart
+void log(String message) {
+  // Log the debug message if the environment declaration 'DEBUG' is `true`.
+  // If there was no value specified, do not log.
+  if (const bool.fromEnvironment('DEBUG', defaultValue: false)) {
+    print('Debug: $message');
+  }
+}
+```
+
+In this snippet, if `DEBUG` is set to `false`
+during compilation, or not specified at all,
+production compilers can completely remove the condition and its body.
+
+The `fromEnvironment` constructors always
+have a default value for when the declaration isn't specified
+or the specified value cannot be parsed.
+Therefore, if you need to specifically check whether
+an environment declaration has been specified,
+use the [`bool.hasEnvironment`][bool-has] constructor:
+
+<?code-excerpt "misc/lib/development/environment_declarations.dart (has-debug)"?>
+```dart
+if (const bool.hasEnvironment('DEBUG')) {
+  print('Debug behavior was configured!');
+}
+```
+
+[string-from]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/String/String.fromEnvironment.html
+[int-from]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/int/int.fromEnvironment.html
+[bool-from]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/bool/bool.fromEnvironment.html
+[bool-has]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/bool/bool.hasEnvironment.html
+
+## Specifying environment declarations
+
+{{site.alert.warning}}
+  Dart tools and compilers currently do not
+  consistently handle environment declarations
+  with comma-separated values.
+  For more details,
+  see [SDK issue 44995][].
```

Comment:
**suggestion:** This could read better. Some additional information from #44495 may make this be more direct.